extension/sigv4authextension/config_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sigv4authextension
+
+import (
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/service/servicetest"
+)
+
+func TestLoadConfig(t *testing.T) {
+	factories, err := componenttest.NopFactories()
+	assert.NoError(t, err)
+
+	factory := NewFactory()
+	factories.Extensions[typeStr] = factory
+	cfg, err := servicetest.LoadConfigAndValidate(path.Join(".", "testdata", "config.yaml"), factories)
+
+	require.NoError(t, err)
+	require.NotNil(t, cfg)
+
+	expected := factory.CreateDefaultConfig().(*Config)
+	expected.Region = "region"
+	expected.Service = "service"
+	expected.RoleArn = "rolearn"
+	expected.STSEndpoint = "regional"
+
+	ext := cfg.Extensions[config.NewComponentID(typeStr)]
+	assert.Equal(t, expected, ext)
+
+	assert.Equal(t, 1, len(cfg.Service.Extensions))
+	assert.Equal(t, config.NewComponentID(typeStr), cfg.Service.Extensions[0])
+}
+
+func TestLoadConfigError(t *testing.T) {
+	factories, err := componenttest.NopFactories()
+	assert.NoError(t, err)
+
+	tests := []struct {
+		name		string
+		expectedErr 	error
+	}{
+		{
+
+			"missing_region",
+			errNoRegionProvided,
+		},
+		{
+			"missing_service",
+			errNoServiceProvided,
+		},
+		{
+			"invalid_endpoint",
+			errInvalidSTSEndpoint,
+		},
+	}
+	for _, testcase := range tests {
+		t.Run(testcase.name, func(t *testing.T) {
+			factory := NewFactory()
+			factories.Extensions[typeStr] = factory
+			cfg, _ := servicetest.LoadConfig(path.Join(".", "testdata", "config_bad.yaml"), factories)
+			extension := cfg.Extensions[config.NewComponentIDWithName(typeStr, testcase.name)]
+			verr := extension.Validate()
```

Comment:
I realized I did not add a test for missing creds here; adding that in, but now I'm a bit confused because a configuration that I thought should provide bad credentials or missing credentials doesn't raise an error. I'm thinking it might be related to the way `getCredsFromConfig()` works, and how currently we don't use `creds.Get()` within it, but I'm not sure.

Reply:
After further testing, it looks like from the configuration, `getCredsFromConfig()` returns an error only if `STSEndpoint` isn't a valid input, in which case `Validate()` will return `errInvalidSTSEndpoint` instead of `errBadCreds`. If I am misunderstanding something about this please let me know, but for now it seems like we could remove `errBadCreds` and the check for `getCredsFromConfig()`, while still setting `cfg.creds = creds`.

Reply:
Resolving this conversation after switching to aws sdk v2